cloud-deployment-guide/cloud-troubleshooting.md

```diff
@@ -0,0 +1,98 @@
+
+## "This domain is not registered with Tiny Cloud. Please see the quick start guide or create an account."
+
+### Cause
+
+This notification is **only shown** when either:
+
+- An `apiKey` is not provided in the script tag.
+- `no-api-key` is provided as the apiKey.
+
+Such as:
+
+```html
+<script src="{{ site.cdnurl }}" referrerpolicy="origin"></script>
+```
+
+### Solution
+
+Update the `src` URL include your (website or application developer's) {{site.cloudname}} API Key. Your API key should replace the string `no-api-key`. For example, if your API is `abcd1234`:
+
+```html
+<script src="https://cdn.tiny.cloud/1/abcd1234/tinymce/5/tinymce.min.js" referrerpolicy="origin"></script>
+```
+
+To retrieve your API key, or to sign up for an API key, visit: [{{site.cloudname}}]({{site.accountsignup}}).
+
+## "The API key you have entered is invalid. Please review your API key."
+
+### Cause
+
+This notification is shown when the API key provided cannot be found on the {{site.cloudname}} server.
+
+The `apiKey` must be:
+
+- An exact length,
+- comprised of certain characters, and
+- created with a {{site.cloudname}} account on the [{{site.accountpage}} page]({{site.accountsignup}}).
+
+### Solution
+
+Check the `apiKey` provided in the script tag:
+
+- Remove any leading or trailing spaces
+- Any other characters that are not in your apiKey. If you are using variable substitution, ensure that the variable is substituting properly.
+- Matches the API key shown at {{site.accountpageurl}}.
+
+## "This domain is not registered with Tiny Cloud. Please review your approved domains."
+
+### Cause
+
+{{site.cloudname}} verifies the domain {{site.productname}} is loading from by checking the domain portion of the **Referer** header in the network request. This notification is shown when the **Referer** of your page does not match the list of approved domains stored against your `apiKey`. You can
+view what your approved domains are in [Account]({{site.accountpageurl}}).
+
+Sometimes the domains in the **Referer** header does not match with the URL in the browser's address bar. To check the **Referer** header, open your browser's _Developer's Tools_ and open the _Network_ tab. From there, find the request being made to load {{site.productname}} with your API key, and click on the **Headers** tab.  In the section called **Request Headers** there should be a field for **Referer**. This is the value that {{site.productname}} is checking against your approved domains. It must match one your approved domains, or you will receive this notification.
+
+### Solution
+
+If the `Referer` is what you are expecting, then you need to ensure that domain is included in your list of approved domains. If the `Referer` is not what you are expecting, you may need to adjust your application's **Referer** header settings.
+
+## "We could not check your domain because the referer header was missing. Please see the guide on how to ensure the referer header is present."
```

Comment:
```suggestion
## "We’re unable to check your domain because the referer header is missing. Please read the Guide on how to ensure your referer header is present, so we can then customize your editor experience."
```